src/main/java/com/avparamonov/checkers/services/GameService.java

```diff
@@ -18,46 +22,84 @@
+
+    private Map<String, Game> currentGames = new HashMap<>();
+
+    public Map<String, Game> getCurrentGames() {
+        return currentGames;
+    }
+
+    public Game createGame(String nickname1, String nickname2) throws PlayerNotFoundException {
+        String gameId = UUID.randomUUID().toString();
+        Player player1 = playerService.findPlayerByNickname(nickname1);
+        Player player2 = playerService.findPlayerByNickname(nickname2);
+
+        if (player1 == null || player2 == null) {
+            throw new ValidationException("Unable to find specified player(s).");
+        }
 
-    public void createGame(Player player1, Player player2) {
         Game game = Game.builder()
-                .board(boardService.initBoard())
-                .players(Arrays.asList(player1, player2))
+                .id(gameId)
+                .board(new Checker[8][8])
+                .player1(player1)
+                .player2(player2)
+                .currentPlayer(player1.getSide() == WHITE ? player1 : player2)
                 .build();
-        gameRepository.save(game);
-
-        List<Checker> redCheckers = game.getBoard().getCells().stream()
-                .filter(c -> c.getChecker() != null)
-                .map(Cell::getChecker)
-                .filter(ch -> ch.getSide() == Side.RED)
-                .collect(Collectors.toList());
-
-        List<Checker> blackCheckers = game.getBoard().getCells().stream()
-                .filter(c -> c.getChecker() != null)
-                .map(Cell::getChecker)
-                .filter(ch -> ch.getSide() == Side.BLACK)
-                .collect(Collectors.toList());
-
-        if (player1.getSide() == Side.RED) {
-            player1.setCheckers(redCheckers);
-            player2.setCheckers(blackCheckers);
-        } else {
-            player2.setCheckers(redCheckers);
-            player1.setCheckers(blackCheckers);
+
+        GameStatus gameStatus = initBoard(game);
+        game.setStatus(gameStatus);
+        currentGames.put(gameId, game);
+
+        return game;
+    }
+
+    private GameStatus initBoard(Game game) {
+        Checker[][] board = game.getBoard();
+
+        for (int row = 0; row < 8; row++) {
+            for (int col = 0; col < 8; col++) {
+                if ( row % 2 == col % 2 ) {
+                    if (row < 3) {
+                        board[row][col] = checkerService.createWithSideAndCoordinates(WHITE, row, col);
+                    }
+                    else if (row > 4) {
+                        board[row][col] = checkerService.createWithSideAndCoordinates(BLACK, row, col);
+                    }
+                    else {
+                        board[row][col] = null;
+                    }
+                }
+                else {
+                    board[row][col] = null;
+                }
+            }
         }
-        playerService.update(player1);
-        playerService.update(player2);
+        return GameStatus.READY;
```

Comment:
Actually I didn't consider to use these statuses during development of this particular issue.
Need to think how and where to use them in general.